### PR TITLE
GraphQL support for Nuclide

### PR DESCRIPTION
**Pull request by zardra**
Added a new page for GraphQL support in Nuclide and update the Outline View and Context View pages to indicate they support GraphQL files.

### CODE REVIEW

**Comment by zardra** [comment]
Hold importing this until an update from Hyo comes through.

**Comment by skevy** [comment]
@zardra was just playing around with this in 0.202.0...needed to make sure that I actually installed GraphQL language service. Might be good to mention that here.

**Comment by skevy** [comment]
There also may be value in writing a quick thing about how to setup GraphQL language service for your project? e.g. requires a .graphqlrc file

**Comment by zardra** [comment]
@skevy That's the information I'm actually waiting for from Hyo. :)

File: docs/_docs/languages/graphql.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Nuclide has built-in support for [GraphQL](http://graphql.org/) using the [GraphQL Language Service](https://github.com/graphql/graphql-language-service).
+
+* TOC
+{:toc}
+
+## Installing GraphQL
+
+Several [server libraries](http://graphql.org/code/) are provided for GraphQL implementation in a wide range of languages.
+
```
Comment (asiandrummer) [inline docs/_docs/languages/graphql.md]: Configuration setup instructions can be written here I believe. I took a pass using information from https://github.com/graphql/graphql-language-service#graphql-configuration-file-graphqlrc - please feel free to modify:

GraphQL Language Service, to provide its full feature set, will need to know some information about your GraphQL development environment. `.graphqlrc` is a GraphQL configuration file that contains this information. `.graphqlrc` can define mutliple configurations for each GraphQL environment, should you have more than one.

Make sure `.graphqlrc` file is configured and saved in your project's top level directory. For more information on how to configure, refer to [this documentation](https://github.com/graphql/graphql-language-service#graphql-configuration-file-graphqlrc).

File: docs/_docs/languages/graphql.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- [Autocomplete](#features__autocomplete)
+- [Go to Definition](#features__go-to-definition)
+- [Outline View](#features__outline-view)
+- [Context View](#features__context-view)
+- [Code Diagnostics](#features__code-diagnostics)
+
+### Autocomplete
+
+Nuclide has access to the schema type information in your project, so autocomplete just works.
```
Comment (asiandrummer) [inline docs/_docs/languages/graphql.md]: Nuclide has access to the schema **with the GraphQL configuration file (`.graphqlrc`)** defined. I'm not sure where to put this information - still wanted to point this out.

**Comment by asiandrummer** [comment]
I should also mention that the community are looking to improve the GraphQL configuration and how-to by suggesting [a generalized approach](https://github.com/graphcool/graphql-config/issues/16). There are still some action items to attend but the suggested JSON format for `.graphqlrc` looks pretty solid. Just wanted to surface this in case we want to include the actual JSON structure in this documentation, in which case we should make sure we update this doc when the language service goes under the change to reflect on the suggested format.

cc @schickling

**Comment by schickling** [comment]
Thanks for looping me in @asiandrummer. Let's try to decide on a first version of the `.graphqlrc` format pretty soon that most tools (like nuclide) can already adopt the format.

File: docs/_docs/languages/graphql.md (hunk 1 above)
Comment (zardra) [inline docs/_docs/languages/graphql.md]: This is great!

**Comment by zardra** [comment]
PR has been amended with suggestions

cc: @asiandrummer

**Comment by wincent** [comment]
@asiandrummer is out traveling this week and he asked me to check in on this. This looks great, @zarda. I'm not sure what the merging process is on the Nuclide repo, but this lg2m.

**Comment by zardra** [comment]
@wincent Thanks! @JoelMarcey should be able to take care of it when he has a chance.

**Pull request closed without merging**